app/models/topic_view_item.rb

```diff
@@ -43,15 +44,30 @@ def self.add(topic_id, ip, user_id = nil, at = nil, skip_redis = false)
 
         result = builder.exec(topic_id: topic_id, ip_address: ip, viewed_at: at, user_id: user_id)
 
-        Topic.where(id: topic_id).update_all "views = views + 1"
-
         if result > 0
           if user_id
             UserStat.where(user_id: user_id).update_all "topics_entered = topics_entered + 1"
           end
         end
 
-        # Update the views count in the parent, if it exists.
+        Topic.where(id: topic_id).update_all "views = views + 1"
+
+        sql = <<~SQL
```

Comment:
This is pretty minor but it feels a bit strange to have this code here but I get why, so we can keep the `TopicViewItem.add` interface unchanged. Maybe we could add a `TopicViewStat.add`, move this code there, and have:

```ruby
Topic.where(id: topic_id).update_all "views = views + 1"
TopicViewStat.add(topic_id: topic_id, viewed_at: at, user_id: user_id)
```

This at least indicates another table is involved now.

Reply:
done